app/assets/javascripts/donut_chart.js

```diff
@@ -0,0 +1,40 @@
+function donut_chart(donut_target_selector, columns, groups, colors) {
+  var donutChartConfig = jQuery().c3ChartDefaults().getDefaultDonutConfig('A');
+  donutChartConfig.bindto = donut_target_selector;
+  donutChartConfig.data = {
+    type: "donut",
+    columns: columns,
+    groups: groups,
+    order: null
+  };
+  donutChartConfig.color = colors;
+  donutChartConfig.tooltip = {
+    contents: function (d) {
+      return '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+        Math.round(d[0].ratio * 100) + '% ' + d[0].name +
+        '</span>';
+    }
+  };
+  c3.generate(donutChartConfig);
+
+  var donutChartTitle = d3.select(donut_target_selector).select('text.c3-chart-arcs-title');
+  donutChartTitle.text("");
+  donutChartTitle.insert('tspan').text(max_percentage(columns) + '%')
+    .classed('donut-title-big-pf', true).attr('dy', 0).attr('x', 0);
+  donutChartTitle.insert('tspan').text(max_label(columns))
+    .classed('donut-title-small-pf', true).attr('dy', 20).attr('x', 0);
+}
+
+function max_percentage(columns) {
+  values = columns.map(function(subarray) { return subarray[1] });
+  max = Math.max.apply(Math, values);
+  sum = values.reduce( (prev, curr) => prev + curr);
```

Comment:
Reminder to self - This is ES6 Javascript and can't be precompiled on production